order-management-service/order-management-server/src/main/java/agh/fis/order_management/component/OrderMapper.java

```diff
@@ -0,0 +1,160 @@
+package agh.fis.order_management.component;
+
+import agh.fis.order_management.model.*;
+import org.aspectj.weaver.ast.Or;
+import org.springframework.stereotype.Component;
+
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class OrderMapper implements IOrderMapper {
+
+    private java.util.Date utilDate = new java.util.Date();
+
+    @Override
+    public List<OrderDto> CreateOrderDtoList(List<OrderEntity> orderEntities) {
+        List<OrderDto> orderDtoList = new ArrayList<>();
+
+        for (OrderEntity orderEntity : orderEntities) {
+            orderDtoList.add(CreateOrderDto(orderEntity));
+        }
+
+        return orderDtoList;
+    }
+
+    @Override
+    public OrderEntity CreateOrderEntity(OrderDto orderDto, int paymentId, OrderStatus status, float price) {
+        OrderEntity orderEntity = new OrderEntity();
+
+        orderEntity.setDate(new Date(utilDate.getTime()));
+        orderEntity.setDelivery(CreateDeliveryEntity(orderDto.getDelivery(), orderEntity));
+        orderEntity.setPaymentMethod(orderDto.getPaymentMethod());
+        orderEntity.setPaymentId(paymentId);
+        orderEntity.setCustomerId(orderDto.getCustomerId());
+        orderEntity.setStatus(status.toString());
+        orderEntity.setPrice(price);
+        orderEntity.setItems(CreateItemEntityList(orderDto.getItems(), orderEntity));
+
+        return orderEntity;
+    }
+
+    @Override
+    public OrderDto CreateOrderDto(OrderEntity orderEntity) {
+        OrderDto orderDto = new OrderDto();
+
+        orderDto.setId(orderEntity.getId());
+        orderDto.setDate(orderEntity.getDate().toString());
+        orderDto.delivery(CreateDelivery(orderEntity.getDelivery()));
+        orderDto.setPaymentMethod(orderEntity.getPaymentMethod());
+        orderDto.setPaymentId(orderEntity.getPaymentId());
+        orderDto.setCustomerId(orderEntity.getCustomerId());
+        orderDto.setStatus(StringToOrderStatus(orderEntity.getStatus()));
+        orderDto.setPrice(orderEntity.getPrice());
+        orderDto.setItems(CreateItemList(orderEntity.getItems()));
+
+        return orderDto;
+    }
+
+    @Override
+    public OrderStatus StringToOrderStatus(String status) {
```

Comment:
Consider storing Starus as its actual type instead of status and create a getter for String type?